From: Chad Dupuis <cdupuis@redhat.com>
Date: Fri, 3 Sep 2010 21:40:04 -0400
Subject: [scsi] qla2xxx: don't {s,g}et port MBC if invalid port id
Message-id: <20100903214004.2767.50140.sendpatchset@localhost.localdomain>
Patchwork-id: 28144
O-Subject: [RHEL 5.6 PATCH 2/9] qla2xxx: Don't issue set or get port param MBC
	if invalid port loop id
Bugzilla: 619814
RH-Acked-by: Rob Evers <revers@redhat.com>

Bugzilla
--------

Bug 619814 (https://bugzilla.redhat.com/show_bug.cgi?id=619814)

Upstream Status
---------------

scsi-misc commit id 17cf2c5d76b468ca03e59c7cf60decfcef6c08c4

Description
-----------

>From 360c6d98362a7b8f8458794b5c1622b981fab8ba Mon Sep 17 00:00:00 2001
From: Madhuranath Iyengar <madhu.iyengar@qlogic.com>
Date: Mon, 28 Jun 2010 12:45:56 +0530
Subject: [PATCH] qla2xxx: Don't issue set or get port param MBC if invalid port loop id


diff --git a/drivers/scsi/qla2xxx/qla_nlnk.c b/drivers/scsi/qla2xxx/qla_nlnk.c
index a25665a..b25e8e8 100644
--- a/drivers/scsi/qla2xxx/qla_nlnk.c
+++ b/drivers/scsi/qla2xxx/qla_nlnk.c
@@ -645,6 +645,14 @@ static int ql_fc_iidma(struct scsi_qla_host *ha, struct sk_buff *skb,
 		goto cleanup;
 	}
 
+	if (fcport->loop_id == FC_NO_LOOP_ID) {
+		DEBUG16(printk(KERN_ERR "%s(%ld): Invalid port loop id, "
+		    "loop_id = 0x%x\n",
+		    __func__, ha->host_no, fcport->loop_id));
+		ql_cmd->error = -EINVAL;
+		goto cleanup;
+	}
+
 	if (port_param->mode)
 		ret = qla2x00_set_idma_speed(ha, fcport->loop_id,
 		    port_param->speed, mb);
